bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationDataPoint.java

```diff
@@ -0,0 +1,306 @@
+
+        return UnDefType.UNDEF;
+    }
+
+    /**
+     * interpret the given byte b and return the value as string.
+     *
+     * @param b
+     * @return sting representation of byte value b in current datatype
+     */
+    public String asString(byte b) {
+        State ste = asState(b);
+        String str = ste.toString();
+        if (ste instanceof UnDefType) {
+            return String.format("<unknown type> %02X ", b);
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * generate byte data to transmit
+     *
+     * @param val is the state of a channel
+     * @return byte value with RS485 representation. Bit level values are returned in the correct location, but other
+     *         bits/datapoints in the same address are zero.
+     */
+    public byte getTransmitDataFor(State val) {
+        byte result = 0;
+        DecimalType value = val.as(DecimalType.class);
+        if (value == null) {
+            /*
+             * if value is not convertible to a numeric type we cannot do anything reasonable with it, let's use the
+             * initial value for it
+             */
+        } else if (datatype == DataType.TEMPERATURE) {
+            int temp = (int) Math.round(value.doubleValue());
+            int i = 0;
+            while (i < Temp_Map.length && Temp_Map[i] < temp) {
+                i++;
+            }
+            result = (byte) i;
+        } else if (datatype == DataType.FANSPEED) {
+            int i = value.intValue();
+            if (i < 0) {
+                i = 0;
+            } else if (i > 8) {
+                i = 8;
+            }
+            result = (byte) Fanspeed_Map[i];
+        } else if (datatype == DataType.BYTE_PERCENT) {
+            result = (byte) ((value.doubleValue() / 100.0) * (255 - BYTE_PERCENT_OFFSET) + BYTE_PERCENT_OFFSET);
+        } else if (datatype == DataType.PERCENT) {
+            double d = (Math.round(value.doubleValue()));
+            if (d < 0.0) {
+                d = 0.0;
+            } else if (d > 100.0) {
+                d = 100.0;
+            }
+            result = (byte) d;
+        } else if (datatype == DataType.HYSTERESIS) {
+            result = (byte) (value.intValue() * 3);
+        } else if (datatype == DataType.SWITCH || datatype == DataType.NUMBER) {
+            // those are the types supporting bit level specification
+            // output only the relevant bits
+            result = (byte) (value.intValue() << bitStart);
+        }
+
+        return result;
+    }
+
+    /**
+     * Get further datapoint linked to the same address.
+     *
+     * @return sister datapoint
+     */
+    public @Nullable HeliosVentilationDataPoint link() {
+        return link;
+    }
+
+    /**
+     * Add a link to a datapoint on the same address.
+     *
+     * @param link is the sister datapoint
+     */
+    @SuppressWarnings("null")
+    public void append(HeliosVentilationDataPoint link) {
+        if (this.link == null) {
```

Comment:
if you check `!=null` and exchange if/else, you can probably remove the `@SuppressWarnings` annotation.

Reply:
Didn't help. Maybe now it's ok...